### ISSUE TITLE
AutoReload database connections

### ISSUE THREAD

**Issue by geferon** [NebulousCloud/helix#59]
Every time any of the gamemode or schema files are modified, autoreload comes into place.
It works very well, but there's a slight problem with the Database.

On the file init.lua (helix/gamemode/init.lua) around line 22~24 the function ix.db.Connect() comes into place.
This connects the server to the database every single time the autoreload comes into place, and therefore, as a chain reaction, it also calls mysql:Connect and, after, mysql:OnConnected, ending on the DatabaseConnected hook, which then calls the hook LoadData, making every single saved entity in the map duplicate.

So, TLDR;
Autoreload causes
ix.db.Connect() -> mysql:Connect() -> mysql:OnConnected() -> DatabaseConnected -> LoadData

I believe that it is unnecesary to reconnect to the database every time the init.lua file is loaded, as with just the first time it's loaded, it should be good to go, so basically the easiest solution would be to add a global variable (probably a boolean) that would be set to true the first time the file got loaded, only running that function once.

**Comment by alexgrist** [comment]
Yeah, I had fixed another instance of this previously. Meant to look into it again after I experienced it again.

There is some cases where you’d want to reconnect the database as sometimes a refresh can happen as it’s timing out so a check for whether it’s already connected will likely suffice.

**Comment by alexgrist** [comment]
Doing that would prevent the initial "connect" when using SQLite as it's considered always connected.

**Comment by MrMe0w** [comment]
Why not connect to the database on the GM:Initialize hook? IIRC it's only called once.
